### DISCUSSION TITLE
[apex] Rules on code

### DISCUSSION THREAD

**Discussion by Puneet1726** [pmd/pmd#3030] [Q&A] [answered]
Hi,

I am planning to use PMD for Apex and one concern raised in the forum was , whether PMD runs analysis locally or code is moved to some other location?

I tried to get the details but not able to find the concrete answer.

Please help me with this question.

Thanks,
Puneet

**Comment by rsoesemann** [comment, marked as answer]
PMD runs locally on the command line. You can also integrate it into all IDE's. If you want to integrate it with you git repo you can connect it to Codacy.com but the it checks not local any more

**Comment by Puneet1726** [comment]
Thanks for the information.

One more question can we segregate issues as critical, blocker, urgent and warning though command line?

I know its possible with eclipse plugin.

Thanks,
Puneet
